feat: d/vsphere_network add support for network_type (#2281)
hashicorp/terraform-provider-vsphere: website/docs/d/network.html.markdown

  network objects, the ID of the distributed virtual switch for which the port
  group belongs. It is useful to differentiate port groups with same name using
  the distributed virtual switch ID.

> * `network_type` - (Optional) For standard virtual switch port groups, distributed virtual switch port groups and NSX port groups.
>   Networks are found by setting `network_type` to `Network`, `DistributedVirtualPortgroup`, or `OpaqueNetwork`. This is required if you have
>   multiple port groups with the same name.
>   

Reviewer: ```suggestion
* `network_type` - (Optional) The managed object type for the discovered network. This is required if you have multiple port groups with the same name. This will be one of `DistributedVirtualPortgroup` for distributed port groups, `Network` for standard (host-based) port groups, or `OpaqueNetwork` for networks managed externally, such as those managed by NSX.
```

  network objects, the ID of the distributed virtual switch for which the port
  group belongs. It is useful to differentiate port groups with same name using
  the distributed virtual switch ID.
* `additional_filters` - (Optional) For standard virtual switch port groups, distributed virtual switch port groups and NSX port groups.

>   Networks are found by setting `network_type` to Network, DistributedVirtualPortgroup or OpaqueNetwork. This is required if you have

Reviewer: ```suggestion
  Networks are found by setting `network_type` to `Network`, `DistributedVirtualPortgroup`, or `OpaqueNetwork`. This is required if you have
```

>   Networks are found by setting `network_type` to Network, DistributedVirtualPortgroup or OpaqueNetwork. This is required if you have

Reviewer: Maybe have this as a supported filters section?

Author: removing filters

>   multiple port groups with the same name

Reviewer: ```suggestion
  multiple port groups with the same name.
```

  network objects, the ID of the distributed virtual switch for which the port
  group belongs. It is useful to differentiate port groups with same name using
  the distributed virtual switch ID.

> * `filter` - (Optional) The type for the discovered network. This is required if you have multiple port groups with the same name. This will be one of `DistributedVirtualPortgroup` for distributed port groups, `Network` for standard (host-based) port groups, or `OpaqueNetwork` for networks managed externally, such as those managed by NSX.

Reviewer: ```suggestion
* `filter` - (Optional) Apply a filter for the discovered network.
   * `network_type`: This is required if you have multiple port groups with the same name. This will be one of `DistributedVirtualPortgroup` for distributed port groups, `Network` for standard (host-based) port groups, or `OpaqueNetwork` for networks managed externally, such as those managed by NSX.
```

Something like this
